### DISCUSSION TITLE
2080 8GB vs 3060 12GB generation speed

### DISCUSSION THREAD

**Discussion by Velour-Fog** [oobabooga/text-generation-webui#875] [Q&A]
I am considering getting a 3060 12GB simply for the VRAM for LLM and diffusion image processing. I currently have an older 2080 8GB and using -wbits 4 --groupsize 128 --pre_layer 35 allows me to run vicuna-13b-GPTQ-4bit-128g at a measly 1 token/s. However I run the much smaller facebook_opt-1.3b at a totally acceptable 17 tokens/s. So I have 2 questions:

1) Is it not being able to fit vicuna-13b-GPTQ-4bit-128g completely in VRAM that makes it run at a snails pace?
2) If I side-grade My card to a 3060 12GB will I see speeds closer to 17/s, OR do different models run a very different speeds (assuming you can fit either in VRAM)?

If anyone out there has a 3060 lemme know. Also, anyone with 12+ GB VRAM, do you see much difference in token/s between different models?

**Comment by Njasa2k** [comment]
I have a 2070, what is your RAM size?

**Reply by Velour-Fog** [reply]
I just re-wrote my comment, but system ram is 32GB.

**Comment by BetaDoggo** [comment]
Larger models require more power and offloading to ram also decreases speeds. If you can fit the entire model into vram with a 3060 the speeds should be much better, but it won't be as fast as the smaller opt1.3b. At your current 1 token per second it would make more sense to use [ggml models](https://github.com/oobabooga/text-generation-webui/wiki/llama.cpp-models) on cpu instead of splitting the hf model.

**Comment by metzo007** [comment]
I have a 3060 12G and it runs vicuna-13b-GPTQ-4bit-128g just fine

I use 
--auto-devices --cai-chat --wbits 4 --groupsize 128

at the start ( I dont know what your --pre_layer 35 does )

I get 6 to 7 tokens per second

**Reply by Jonseed** [reply]
I have the same card, and only get about 3 tokens per second, but that might be because I used the one-click installer on Windows, instead of going through WSL. Did you use WSL?

**Reply by Felixqian4160** [reply]
ME too I have a 3060 12G , my speed is similar like your speed, use the same models

**Reply by mrharrison007** [reply]
I did the one clicker, 3060 12G, old 17 32Gb.

To create a public link, set `share=True` in `launch()`.
Output generated in 6.86 seconds (1.90 tokens/s, 13 tokens, context 35, seed 1516750171)
Output generated in 24.99 seconds (5.76 tokens/s, 144 tokens, context 74, seed 1656027071)
Output generated in 7.55 seconds (5.30 tokens/s, 40 tokens, context 234, seed 1322186380)
Output generated in 3.77 seconds (3.18 tokens/s, 12 tokens, context 297, seed 1146708843)
Output generated in 6.36 seconds (4.41 tokens/s, 28 tokens, context 338, seed 2007227002)
Output generated in 9.46 seconds (4.97 tokens/s, 47 tokens, context 384, seed 398346280)
Output generated in 2.90 seconds (1.38 tokens/s, 4 tokens, context 427, seed 472241879)
Output generated in 6.19 seconds (5.82 tokens/s, 36 tokens, context 41, seed 573578846)
Output generated in 12.23 seconds (5.56 tokens/s, 68 tokens, context 95, seed 157178054)
Output generated in 4.72 seconds (4.45 tokens/s, 21 tokens, context 44, seed 2016777684)

**Comment by El-Hondo** [comment]
Running a 2080Ti here. Got mine second hand for stable diffusion stuff. It runs pretty fast. Only the 11GB could be a concern in the future.

**Comment by marcoseduardopm** [comment]
> I am considering getting a 3060 12GB simply for the VRAM for LLM and diffusion image processing. I currently have an older 2080 8GB and using -wbits 4 --groupsize 128 --pre_layer 35 allows me to run vicuna-13b-GPTQ-4bit-128g at a measly 1 token/s. However I run the much smaller facebook_opt-1.3b at a totally acceptable 17 tokens/s. So I have 2 questions:
> 
> 1. Is it not being able to fit vicuna-13b-GPTQ-4bit-128g completely in VRAM that makes it run at a snails pace?
> 2. If I side-grade My card to a 3060 12GB will I see speeds closer to 17/s, OR do different models run a very different speeds (assuming you can fit either in VRAM)?
> 
> If anyone out there has a 3060 lemme know. Also, anyone with 12+ GB VRAM, do you see much difference in token/s between different models?

Go ahead and buy the second card if your motherboard&power supply support 2 GPUs. text-generation-webui runs perfectly fine with 2 different GPUs using the argument --gpu-memory followed by the memory of each card. In that case, you will need to remove the argument --pre_layer 35 (which severely lowers your performance but saves VRAM), and your cards together will have the memory necessary for the vicuna-13b-GPTQ-4bit-128g model.

**Comment by cduk** [comment]
Yes, offloading parts to the CPU really slows things down. I got a 2080 TI with 11GB and if the model fits into VRAM, I get around 17 tokens/s. Now I want to upgrade to be able to use even bigger models.

**Comment by Felixqian4160** [comment]
You can buy second card like 2080ti 22G ,this card almost like 3090

**Reply by cduk** [reply]
Yes, I was wondering about a 2nd card and how much performance penalty there is compared to a single 24GB card.

**Reply by Felixqian4160** [reply]
you can see the Assessment videos about 2080ti 22G

**Reply by cduk** [reply]
At first, I thought the 22G was a typo, but I saw that there are modded 2080TIs for sale. I would have loved to go down this route, but the seller doesn't ship to my country.

**Comment by kuso-ge** [comment]
I also have a 3060 12GB GPU.
At context 0-400, I get around 9-10.4 tokens/s
at max context 1800~, I drop down to 3-3.7 tokens/s
Although from time to time, when my context is too high, the generation ends prematurely without reaching the MAX_NEW_TOKENs I set(usually between 200-600, if you set it too high the context basically disappears).  So I just disable the EOS token so that the generation would cap at the maximum token limit.

**Comment by mzbac** [comment]
I am looking to buy a 3060 but now I'm wondering if it's worth it. Currently, I am running the stable-vicuna-13B.ggml.q5_1.bin on an i7 7820x with 80GB of RAM and getting around 4 tokens per second.
```
Output generated in 48.55 seconds (4.12 tokens/s, 200 tokens, context 39, seed 67842338)
Output generated in 45.81 seconds (4.37 tokens/s, 200 tokens, context 24, seed 1880916522)
Output generated in 167.73 seconds (4.44 tokens/s, 744 tokens, context 24, seed 2040648477)
Output generated in 438.58 seconds (4.10 tokens/s, 1800 tokens, context 28, seed 911192468)
```

**Reply by Felixqian4160** [reply]
The power consumption and cost of 3060 12G and i7 80G, how much difference is there, it should still be 3060 12G’s cost performance is higher, but I think 2080ti22G should be better.

**Reply by mzbac** [reply]
Just buy a second-hand workstation, such as an HP Z4 G4, and add 32GB of RAM. It should cost around $300-$400, plus you don't need a GPU and consistent speed of around 4 tokens/s.

**Reply by mzbac** [reply]
I just got 3060 12 today with 13b stable vicuna GPTQ model. It is consistently producing 7 t/s. It doesn't seem like there has been much improvement. maybe 4070 is way to go :p

**Comment by valden80** [comment]
I'm have 3060 with 12GB vram. I run it on linux, with no display, exclusive for CUDA tasks (My display and X.Org use intel video of CPU, so all 12GB of 3060 is available to cuda.) This way I can use almost any 4bit 13b llama-based model, and full 2048 context, at regular speed up to ~15 t/s. (Speed may be varied from model to model and state of context, but no less then 6-8 t/s). Main exclusion is one model - Erebus 13b 4bit, that I found somewhere at huggingface. It can be loaded too, but generate very slowly  ~1 t/s at most. 
But for 7B models even small sized Lora training is possible in 4bit mode.

**Comment by R4X70N** [comment]
I am running dual NVIDIA 3060 GPUs, totaling 24GB of VRAM, on Ubuntu server in my dedicated AI setup, and I've found it to be quite effective. When running smaller models or utilizing 8-bit or 4-bit versions, I achieve between 10-15 tokens/s. It's possible to run the full Vicuna-13b model as well, although the token generation rate drops to around 2-3 tokens/s and consumes about 22GB out of the 24GB of available VRAM. Nonetheless, it does run.

In my case, I chose to prioritize running larger models at a slower pace rather than focusing on faster performance with smaller models. I'm quite satisfied with this decision and can attest to the effectiveness of LLAMA's dual GPU support. This could be a viable option for you to consider, as opposed to investing in a single large card.
Its surprising how little information is out there on this setup

**Reply by Velour-Fog** [reply]
Hey there, OP here, I was also considering getting 2 3060s as well, but was concerned about any annoyances involved with a dual GPU setup running these programs. Is it as straightforward as installing 2 GPUs and these popular programs such as oobabooga and Automatic1111 etc. will recognize them, and seamlessly utilize the VRAM across both GPUs?

**Reply by R4X70N** [reply]
I am using Ubuntu Server and I haven't had many issues. If you are using Windows I don't know. There are always going to be some issues and troubleshooting when using a special setup but I haven't had any major issues. 

I've only really messed around with Oobabooga, It has been running great, and it has fantastic support for multi-GPU. I recommend starting the WebUI without any models loaded and then loading them manually from the model page. This way you can change the settings before anything tries to load. The settings are straightforward but it will take some trial and error.
An issue I have had is that the GPTQ quantized models don't always play nice with both GPUs and I have no idea why. Sometimes it will ignore one of the GPUs and sometimes it's fine.
If you do end up using two I recommend setting the second GPUs memory higher than the first because the memory and context seem to be stored on the first so it needs room to grow.

Overall being able to run most 13b models either full 16-bit or 8-bit outweighs the slower token speeds. I can run StableVicuna 13b in 8-bit and it runs at about 5 tokens/s which isn't as slow as you would think.

**Reply by kuso-ge** [reply]
been wanting to try a dual GPU setup as well. Sadly, I have an mtx mobo and while I do have an extra PCIE, it's impossible to fit in 2 GPU's in my case. I would either have to upgrade my mobo or go Caseless.

**Reply by R4X70N** [reply]
Make sure that if you are going to use two cards you need the second PCIe slot to be an 8x slot. It might look like a 16x but it might not have all of the pins. Nvidia cards need at least an 8x slot while AMD cards only need a 4x.
